nx/test/nx/defn/grad_test.exs

```diff
@@ -1580,6 +1574,57 @@ defmodule Nx.Defn.GradTest do
     end
   end
 
+  describe "concatenate" do
+    defn concatenate_grad(t) do
+      grad(t, Nx.sum(Nx.concatenate([t, t], axis: 0)))
+    end
+
+    defn concatenate_grad_power(t) do
+      grad(
+        t,
+        Nx.sum(Nx.concatenate([Nx.power(t, 2), Nx.power(t, 3)], axis: 0))
+      )
+    end
```

Comment:
If I remove all sums I get a "** (ArgumentError) can only compute gradients of expressions that return scalars, got shape: {2}" message. I assume the issue you commented about was that the outer sum was a no-op?

Other than this, there seems to be something wrong with the grad when applied to the `Nx.power` calls.
The expected result is still documented in the relevant test assertion, but the actual result is `Nx.tensor([5.0, 112.0], [783.0, 3.2e3]])`. I'll keep investigating

Reply:
Sum should sum across all dimensions by default and return a scalar, no?

Reply:
Yes, that's what I meant by "no-op": sum(sum(....sum(x))) === sum(x) when using the default options. At first I thought that the sum would happen across a single axis, and thus the need for nesting sums.